src/controllers/employeeDetails.controller.js

```diff
@@ -24,6 +24,39 @@ const {employeeValidation,getId,getListValidation,recoveryParamsValidation,updat
     }
   };
 
+  //create Employee details with condition
+  const addOne = async (req ,res ,next) => {
+    const reqData = { ...req.query };
+    try {
+    const { employee_id, 
+      department_id, 
+      first_name, 
+      last_name, 
+      address,
+      email_id, 
+      phone_no,}  = await employeeValidation.validateAsync(req.body);
+    const { page_size,page_no } = await getListValidation.validateAsync(reqData);
+    const department = await departmentService.getAll({ page_size, page_no });
+    console.log('-------->>>>',department);
+    if( !department_id ){
+      throw (`department id ${req.body.department_id} doest not exist`);
+    }else{
+    const employee = await employeeDetailsService.addOne({ employee_id,department_id,first_name, last_name, email_id, phone_no, address,})
+    return success.handler({ employee }, req, res, next);
+    }
+    }catch (err) {
+      switch (err.name) {
+        case 'SequelizeUniqueConstraintError':
+          err.custom_key = 'employeeDetailsConflict';
+          err.message = `department id  ${req.body.department_id} does not exists`;
```

Comment:
If you have already extracted the department_id, use it directly instead of extracting it again from body.